SplunkAppForWazuh/appserver/static/js/controllers/overview/aws/awsCtrl.js

```diff
@@ -2,74 +2,156 @@ define([
+        }
+      } else {
+        this.awsMetrics = this.awsMetrics[0]['aws-s3']
+        this.scope.awsMetrics = {
+          enabled: "",
+          scanInterval: this.awsMetrics.interval,
+          runOnStart: this.awsMetrics.run_on_start,
+          skipOnError: this.awsMetrics.skip_on_error,
+          buckets: "",
+          accountInUse: "",
+          regionsInUse: ""
+        }
+        if (this.awsMetrics.disabled == "no")
+          this.scope.awsMetrics.enabled = "yes"
+        else
+          this.scope.awsMetrics.enabled = "no"
+        this.awsMetrics.buckets.map(bucket => {
+          this.scope.awsMetrics.buckets = `${this.scope.awsMetrics.buckets} ${bucket.name}`
+        })
+        this.serializedAwsMetrics = this.serializedMetrics(this.awsMetrics.buckets)
+        this.scope.awsMetrics.regionsInUse = this.serializedAwsMetrics.regions
+        this.scope.awsMetrics.accountInUse = this.serializedAwsMetrics.accounts
+      }
+
+      this.dropdown = new Dropdown(
+        'dropDownInput',
+        `${this.implicitFilters} sourcetype=wazuh data.aws.source=* | stats by data.aws.source | fields data.aws.source | sort data.aws.source ASC`,
+        'data.aws.source',
+        '$form.awsSource$',
+        'dropDownInput'
+      )
+      this.dropdownInstance = this.dropdown.getElement()
+      this.dropdownInstance.on('change', (newValue) => {
+        try {
+          this.oldValue = window.localStorage.getItem('lastAwsSourceFilter')
+          if (!this.oldValue || (this.oldValue && this.oldValue != newValue)) {
+            this.awsCurrentFilters = []
+            if (JSON.parse(window.localStorage.getItem('awsSourceFilters')))
+              this.awsCurrentFilters = JSON.parse(window.localStorage.getItem('awsSourceFilters'))
+            this.sourceValues = this.getAwsFiltersValue()
+            this.newKey = 'data.aws.source'
+            this.newFilter = {}
+            this.newFilter[this.newKey] = newValue
+            this.oldValue = window.localStorage.getItem('lastAwsFilter')
+            if (newValue === '*') {
+              this.newAwsFilters = this.awsCurrentFilters.filter(item => !item['data.aws.source'])
+              window.localStorage.setItem('awsSourceFilters', JSON.stringify(this.newAwsFilters))
+              window.localStorage.setItem('lastAwsSourceFilter', newValue)
+              this.rootScope.$broadcast('applyFiltersAws')
+              this.state.reload()
+            } else {
+              if (!this.sourceValues.includes(newValue)) {
+                this.awsCurrentFilters.push(this.newFilter)
+                window.localStorage.setItem('awsSourceFilters', JSON.stringify(this.awsCurrentFilters))
+                window.localStorage.setItem('lastAwsSourceFilter', newValue)
+                this.rootScope.$broadcast('applyFiltersAws')
+                this.state.reload()
+              }
+            }
+          }
+        } catch (err) {
+          console.error("error: ", err)

```

Comment:
This is useless, please do something with the error instead of sending it to the console